logic/tasks/ProgressProvider.h

```diff
@@ -23,6 +23,8 @@ class ProgressProvider : public QObject
 protected:
 	explicit ProgressProvider(QObject *parent = 0) : QObject(parent)
 	{
+		connect(this, &ProgressProvider::succeeded, this, &ProgressProvider::finished);
```

Comment:
There is indeed. There are cases when you don't care how something finished, you just care about it being done.